### ISSUE TITLE
Version 1.0.3 - Somewhat broken on Samsung Galaxy S6

### ISSUE THREAD

**Issue by kentskoglund** [Yubico/yubioath-android#29]
Latest update broke it Latest update 1.0.3 broke it for me. When I try to scan the yubikey, it only returns "error in yubikey communication". Makes no difference is the app is already open or not. Worked fine in 1.0.2.

Stock Samsung galaxy S6, SM-G920F
Android 5.1.1
Build LMY47X.G920FXXU3COI9

Update:
Did some testing, and I'm getting mixed results. This is purely an amateurs  fiddling, but here goes.

Version 1.0.2 worked to a near 100%. After updating to 1.0.3, I was unable to retrieve my codes, even after 15-20 attempts. After submitting this Github issue, I tried several things. 
1. Cleared app cash and data, forced stopped the app and tried to read the yubikey 4-5 times. No luck.
2. Cleared app cash and data, uninstalled the app and reinstalled it. Tried to read the yubikey 4-5 times. No luck
3. Forced closed the app, and tried the yubikey directly from the homescreen (which I never usually do) and heard the nfc success sound and was shown the "choose standard app" menu. I chose the Yubico Auth app with the "Just Once" option. The app opened, but no codes. But then i tried the yubikey again with the app opened, and this time I heard the nfc failure sound but it worked and the codes were shown. Now I no longer get the "choose app" menu when trying from the homescreen, and it never takes me to the Yubico Auth app, but i does give the nfc success sound. But if I open the app and try from there, it works about 20% of the time even though I always get the nfc failure sound. It's working in a manner, but in a far less reliable way than the 1.0.2 version.

**Comment by dainnilsson** [comment]
Thanks for adding your test results. It seems related to the switch to using the Nordpol library for NFC communication. The irony is that using this library is intended to make the NFC work _better_ for Samsung devices, and here it seems to have made it worse. @petter-fidesmo, @Franzaine, is this anything you've run across before? Have you done any testing with an S6 device?

**Comment by Franzaine** [comment]
@kentskoglund Thank you very much for your test results!

@dainnilsson This is a truly weird error. We have a S6 with the exact same config (S6, SM-G920F, Android 5.1.1, Build LMY47X.G920FXXU3COI9) in the office and are unable to reproduce this behaviour. I'll test with our Yubikey NEO as soon as I get access to it (it's out travelling ;) ). Should be back tomorrow.

@petter-fidesmo has a theory that it could possibly be the NDEF on the Yubikey that "spooks" the NFC stack in reader mode.

**Comment by petter-fidesmo** [comment]
We have been doing a lot of testing here today, with two Yubikeys of different versions and some Fidesmo cards. With the newest Yubikey it still works quite nice for us with the S6. With the older one it works a bit worse. The main issue seems to be that the tag is very often lost by the phone (possibly due to its small antenna). In the 1.0.2 mode the tag was lost even before it was forwarded to the app, while with nordpol (that is in reader mode) it is picked up quicker and the failure instead happens when the app talks to the tag. So failure rate seems to be the same in the two version, but in the new version it leads to an application failure. We are currently looking at what would be the best work around for this and if the app can work well with both the S5 and S6 at the same time. Fidesmo cards works very well with both the new and old version, probably due to their big antenna.

We hope to have some updates early next week.

**Comment by petter-fidesmo** [comment]
Please see #35 for a possible fix to this problem.

**Comment by kentskoglund** [comment]
Tried the apk http://nordpol.fidesmo.com.s3-website-eu-west-1.amazonaws.com/yubioath.apk and the results of my testing is mixed. Now everytime I try the Yubikey, I get the success sound and no error message. But the codes will only show up in about 1 of 5 attempts as before. 

Might add that my Yubikey Neo has firmware v. 3.3. 

Think I'll try compiling the 1.0.2 version and use that for now.

**Comment by dainnilsson** [comment]
If you need an older version of the app you can also get it here: https://developers.yubico.com/yubioath-android/Releases/

**Comment by kentskoglund** [comment]
Have tried the older versions now, and I'm getting the same results!? Is it the phone that has changed? Every older version I try, gives me the same results. 

I tried with my backup Yubikey Neo, bought at the same time, but there is no difference.

I bought a brand new Yubikey Neo, thinking it might be good to have a newer key anyway, but Yubikey managed to send me the Yubikey 4 and not the Yubikey Neo I ordered :(

**Comment by dainnilsson** [comment]
Hi, sorry to hear that. Please contact our support (https://www.yubico.com/support/) regarding the wrong shipment and they will resolve that.

**Comment by dainnilsson** [comment]
Closing this issue now, as I'm not sure it's applicable to the current version.

**Comment by kenydub** [comment]
I just purchased a yubikey neo and have been trying to get it to work with a Samsung Galaxy S6 without success. The key works with Trigger - Task Launcher (https://play.google.com/store/apps/details?id=com.jwsoft.nfcactionlauncher&hl=en) on the Samsung but with the authenticator I get "error in yubikey communication".  I have tried the most recent version of the authenticator  and earlier version at the site above again without success.  Android version is 7.0 and build number is NRD90M.G920VVRS4DQH1. Any suggestions of what I can do to get this to work? Thanks.
